src/resources/js/components/FormatForm.svelte

```diff
@@ -0,0 +1,257 @@
+      } else {
+        savedFormat = await RootServerApi.createFormat({ worldId: values.worldId, type: values.type, translations: translations });
+      }
+    },
+    onError: async (error) => {
+      console.log(error);
+      await RootServerApi.handleErrors(error as Error, {
+        handleValidationError: (error) => {
+          const errorObject: any = {};
+          for (const lang of allLanguages) {
+            const k = error?.errors[lang + '_key'] ?? [];
+            // If there is a name or description but the key is missing, note that there is an error.  (To avoid a
+            // circularity, this isn't done in yup, which would otherwise be the logical place for this check.)
+            if (!$data[lang + '_key'] && ($data[lang + '_name'] || $data[lang + '_description'])) {
+              k.push($translations.keyIsRequired);
+            }
+            errorObject[lang + '_key'] = k.join(' ');
+            const n = error?.errors[lang + '_name'] ?? [];
+            errorObject[lang + '_name'] = n.join(' ');
+            const d = error?.errors[lang + '_description'] ?? [];
+            errorObject[lang + '_description'] = d.join(' ');
+          }
+          errors.set(errorObject);
+        }
+      });
+      spinner.hide();
+    },
+    onSuccess: () => {
+      spinner.hide();
+      dispatch('saved', { format: savedFormat });
+    },
+    extend: validator({ schema: yup.object(yupSchema), castValues: true })
+  });
+
+  // This hack is required until https://github.com/themesberg/flowbite-svelte/issues/1395 is fixed.
+  function disableButtonHack(event: MouseEvent) {
+    if (!$isDirty) {
+      event.preventDefault();
+    }
+  }
+
+  // Hack to provide a label for the key, name, and description fields in all languages -- if it's not defined
+  // in the current language, use English.  TODO: maybe get rid of this if these are defined for all languages?
+  function getLabel(title: string, lang: string): string {
+    const t = $translations.getString(title, lang, true);
+    return t ? t : $translations.getString(title, 'en');
+  }
+
+  $: isDirty.set(formIsDirty(initialValues, $data));
+
+  // TODO: the form below has an 'if' to expand the accordion item just for the current language.  This inelegantly repeats the
+  // accordion item contents. Either figure out how to conditionally have either '<AccordionItem open>' or just '<Accordion>',
+  // or else factor out the item contents into a separate component.
+</script>
+
+<form use:form>
+  <div class="grid gap-4 md:grid-cols-2">
+    {#if selectedFormat?.id}
+      <div class="text-gray-700 md:col-span-2 dark:text-gray-300">
+        <strong>{$translations.formatId}:</strong>
+        {selectedFormat?.id}
+      </div>
+    {/if}
+    <div class="md:col-span-2">
+      <Accordion multiple>
+        {#each allLanguages as lang}
+          {#if $translations.getLanguage() === lang}
+            <AccordionItem open>
```

Comment:
This code should be simplified to remove the duplication.  Is there a way to make the open argument conditional on `$translations.getLanguage() === lang}`?  If not, the duplication could be removed by factoring out the `AccordionItem` contents into a separate component.

Reply:
Fixed with #1059